### PR TITLE
helm: Ensure proper labels on all resources

### PR DESCRIPTION
**Pull request by wtty-fool**
Towards https://github.com/giantswarm/giantswarm/issues/8632

Mirrors changes made by https://github.com/giantswarm/aws-operator/pull/2171.

### CODE REVIEW

File: helm/cluster-operator/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Common labels
+*/}}
+{{- define "cluster-operator.labels" -}}
+helm.sh/chart: {{ include "cluster-operator.chart" . | quote }}
+app: {{ include "cluster-operator.name" . | quote }}
+{{ include "cluster-operator.selectorLabels" . }}
+app.giantswarm.io/branch: {{ .Values.project.branch | quote }}
+app.giantswarm.io/commit: {{ .Values.project.commit | quote }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
```
Comment (kopiczko) [inline helm/cluster-operator/templates/_helpers.tpl]: Alphabet please.

File: helm/cluster-operator/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+{{- end -}}
+
+{{/*
+Selector labels
+*/}}
+{{- define "cluster-operator.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "cluster-operator.name" . | quote }}
+app.kubernetes.io/instance: {{ .Release.Name | quote }}
```
Comment (kopiczko) [inline helm/cluster-operator/templates/_helpers.tpl]: Alphabet.

File: helm/cluster-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -4,21 +4,18 @@ metadata:
...
 spec:
   replicas: 1
   selector:
     matchLabels:
-      app: {{ .Values.project.name }}
-      version: {{ .Values.project.version }}
+      {{- include "cluster-operator.selectorLabels" . | nindent 6 }}
   strategy:
-    type: Recreate
+    type: RollingUpdate
```
Comment (kopiczko) [inline helm/cluster-operator/templates/deployment.yaml]: Nope. We want `Recreate`. Why did that change?

Comment (wtty-fool) [reply helm/cluster-operator/templates/deployment.yaml]: It was changed in giantswarm/aws-operator/pull/2171 before. Reverting.

File: helm/cluster-operator/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+{{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "cluster-operator.labels" -}}
+app: {{ include "cluster-operator.name" . | quote }}
+app.giantswarm.io/branch: {{ .Values.project.branch | quote }}
+app.giantswarm.io/commit: {{ .Values.project.commit | quote }}
+{{ include "cluster-operator.selectorLabels" . }}
```
Comment (kopiczko) [inline helm/cluster-operator/templates/_helpers.tpl]: I wouldn't go as crazy to sort `include` contents :D Just put it on the top or the bottom :)

Comment (wtty-fool) [reply helm/cluster-operator/templates/_helpers.tpl]: Well, I did :sweat_smile: 
Ok

**Pull request merged**